packages/engine/src/datastore/table/context/mod.rs

```diff
@@ -94,7 +103,10 @@ impl Context {
     }
 
     pub fn into_pre_context(self) -> PreContext {
-        PreContext { batch: self.batch }
+        PreContext {
+            batch: self.batch,
+            removed_batches: self.removed_batches,
```

Comment:
```suggestion
            // TODO: OPTIM - We should be using these to release memory, this requires propagation to the 
            // runners, otherwise this is the cause of a possible memory leak
            removed_batches: self.removed_batches,
```

Can you also open an asana task?

Reply:
I added this to the corresponding method: https://github.com/hashintel/hash/commit/21d0819fbca1de7a062b6fa188c2b81895858248

[Asana task](https://app.asana.com/0/1199548034582004/1201940767289027/f) (_internal_)

Reply:
Ah I looked for it but missed it. Can you replace the TODO with the one I've suggested.

Reply:
https://github.com/hashintel/hash/pull/385/commits/84bc6e7766aaac25bfe015ae2f56254437024883

Reply:
There are two parts to freeing memory -- unmapping and freeing loaded data -- so the task should mention both of them or there should be two tasks.

Reply:
Feel free to edit the task to make it clearer :)